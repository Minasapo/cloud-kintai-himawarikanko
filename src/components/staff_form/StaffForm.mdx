import {
  Canvas,
  Meta,
  Title,
  Subtitle,
  Description,
  Primary,
  Controls,
  Stories,
  Source,
  Story,
} from "@storybook/blocks";

import * as StaffFormStories from "./StaffForm.stories";

<Meta of={StaffFormStories} />

# StaffForm

アプリケーション内で使用するスタッフのフォームです。\
具体的なユースケースが決まっているものは、ストーリーに記載されています。

<Primary />

# パラメーター

コンポーネント内で使用できるパラメーターです。\
パラメーターを変更することで上にあるコンポーネントが変化します。

<Controls />

# ストーリー

コンポーネントのユースケースを記載しています。\
既存のコンポーネントを使用する際に参考にしてください。

## スタッフ作成

システム管理者とスタッフ管理者のいずれもスタッフを作成できます。\
システム管理者のみ権限で`システム管理者`を選択できます。

### システム管理者

<Canvas of={StaffFormStories.SystemAdminCreateStaff} />

### スタッフ管理者

<Canvas of={StaffFormStories.StaffAdminCreateStaff} />

## スタッフ更新

システム管理者とスタッフ管理者のいずれもスタッフを更新できます。\
システム管理者のみ権限で`システム管理者`を選択できます。

### システム管理者

<Canvas of={StaffFormStories.SystemAdminUpdateStaff} />

### スタッフ管理者

<Canvas of={StaffFormStories.StaffAdminUpdateStaff} />
