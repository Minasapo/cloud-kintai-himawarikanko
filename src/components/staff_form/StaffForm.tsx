// cspell:words testid
import {
  Box,
  FormControl,
  InputLabel,
  MenuItem,
  Select,
  SelectChangeEvent,
  Stack,
  TextField,
} from "@mui/material";
import { useEffect, useRef, useState } from "react";
import { useAppDispatch, useAppSelector } from "../../lib/hooks";
import createStaff from "../../lib/staff/CreateStaff";
import updateStaff from "../../lib/staff/UpdateStaff";
import updateStaffRole from "../../lib/staff_role/updateStaffRole";
import { selectLoginStaff, selectStaffList } from "../../lib/store";
import Button from "../button/Button";

const StaffForm = () => {
  const loginStaff = useAppSelector(selectLoginStaff);
  const staffs = useAppSelector(selectStaffList);

  const dispatch = useAppDispatch();
  const selectedStaff = staffs.selectedData;

  const [lastName, setLastName] = useState(selectedStaff?.lastName || "");
  const [firstName, setFirstName] = useState(selectedStaff?.firstName || "");
  const [mailAddress, setMailAddress] = useState(
    selectedStaff?.mailAddress || ""
  );
  const [selectedRole, setSelectedRole] = useState(
    selectedStaff?.staffRoles.roleId || 2
  );
  const [saveButtonDisabled, setSaveButtonDisabled] = useState(true);
  const [createButtonDisabled, setCreateButtonDisabled] = useState(true);

  const mailAddressInputRef = useRef<HTMLInputElement>(null);

  const [mailAddressInputError, setMailAddressInputError] = useState(false);

  const handleChangeRole = (event: SelectChangeEvent) => {
    setSelectedRole(Number(event.target.value));
  };

  const handleChangeLastName = (event: React.ChangeEvent<HTMLInputElement>) => {
    setLastName(event.target.value);
  };

  const handleChangeFirstName = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    setFirstName(event.target.value);
  };

  const handleChangeMailAddress = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    if (mailAddressInputRef.current) {
      const ref = mailAddressInputRef.current;
      setMailAddressInputError(!ref.validity.valid);

      if (!ref.value) {
        setMailAddressInputError(false);
      }
    }
    setMailAddress(event.target.value);
  };

  const handleSave = () => {
    if (selectedStaff) {
      void dispatch(updateStaff({ staff: selectedStaff }));
      void dispatch(
        updateStaffRole({
          staffId: selectedStaff.staffId,
          roleId: selectedRole,
        })
      );
    }
  };

  const handleCreate = () => {
    void dispatch(
      createStaff({
        staffCreate: {
          lastName,
          firstName,
          mailAddress,
        },
      })
    );

    setCreateButtonDisabled(true);
  };

  useEffect(() => {
    setLastName(selectedStaff?.lastName || "");
    setFirstName(selectedStaff?.firstName || "");
    setMailAddress(selectedStaff?.mailAddress || "");
    setSelectedRole(selectedStaff?.staffRoles.roleId || 2);
  }, [selectedStaff]);

  useEffect(() => {
    if (selectedStaff) {
      const isDisabled =
        lastName === "" ||
        firstName === "" ||
        mailAddress === "" ||
        (selectedStaff.lastName === lastName &&
          selectedStaff.firstName === firstName &&
          selectedStaff.mailAddress === mailAddress &&
          selectedStaff.staffRoles.roleId === selectedRole) ||
        mailAddressInputError;
      setSaveButtonDisabled(isDisabled);
      return;
    }

    const isDisabled =
      lastName === "" ||
      firstName === "" ||
      mailAddress === "" ||
      mailAddressInputError;

    setCreateButtonDisabled(isDisabled);
  }, [lastName, firstName, mailAddress, selectedRole]);

  return (
    <Stack spacing={1}>
      {staffs.selectedData && (
        <Box
          sx={{
            display: "inline-flex",
            flexDirection: "row",
            justifyContent: "flex-end",
          }}
        >
          <Button color="delete" label="スタッフを削除" />
        </Box>
      )}
      <Box>
        <Stack spacing={3} sx={{ display: "inline-block" }}>
          <Box>
            <TextField
              label="スタッフコード"
              value={selectedStaff?.staffId || ""}
              disabled={true}
            />
          </Box>
          <Box>
            <Stack direction="row" spacing={3}>
              <Box>
                <TextField
                  label="名前(姓)"
                  inputProps={{ maxLength: 10, "data-testid": "last-name" }}
                  value={lastName}
                  required={true}
                  onChange={handleChangeLastName}
                />
              </Box>
              <Box>
                <TextField
                  label="名前(名)"
                  inputProps={{ maxLength: 10, "data-testid": "first-name" }}
                  value={firstName}
                  required={true}
                  onChange={handleChangeFirstName}
                />
              </Box>
            </Stack>
          </Box>
          <Box>
            <TextField
              label="メールアドレス"
              inputRef={mailAddressInputRef}
              inputProps={{
                type: "email",
                maxLength: 100,
                pattern: "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+.[a-zA-Z]{2,}$",
                "data-testid": "mail-address",
              }}
              error={mailAddressInputError}
              helperText={
                mailAddressInputError ? "入力内容に誤りがあります" : ""
              }
              value={mailAddress}
              required={true}
              onChange={handleChangeMailAddress}
              sx={{ width: 1 }}
            />
          </Box>
          <Box>
            <FormControl fullWidth>
              <InputLabel required={true}>権限</InputLabel>
              <Select
                data-testid="staff-role"
                labelId="staff-role"
                label="権限"
                value={String(selectedRole)}
                required={true}
                sx={{ width: 1 }}
                onChange={handleChangeRole}
              >
                {loginStaff.data?.staffRoles.roleId === 1 && (
                  <MenuItem value={1}>システム管理者</MenuItem>
                )}
                <MenuItem value={2}>スタッフ</MenuItem>
                <MenuItem value={3}>スタッフ管理者</MenuItem>
              </Select>
            </FormControl>
          </Box>
          <Box>
            {staffs.selectedData ? (
              <Button
                label="保存"
                disabled={saveButtonDisabled}
                onClick={handleSave}
              />
            ) : (
              <Button
                label="作成"
                disabled={createButtonDisabled}
                onClick={handleCreate}
              />
            )}
          </Box>
        </Stack>
      </Box>
    </Stack>
  );
};
export default StaffForm;
