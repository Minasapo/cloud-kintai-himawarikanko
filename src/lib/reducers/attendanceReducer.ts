import { ActionReducerMapBuilder, createSlice } from "@reduxjs/toolkit";
import fetchAttendance, {
  OriginAttendance,
} from "../time_record/FetchAttendance";
import registerClockIn from "../time_record/RegisterClockIn";
import registerClockOut from "../time_record/RegisterClockOut";
import updateRemarks from "../time_record/UpdateRemarks";

export enum AttendanceStatus {
  NOT_PROCESSING = "NOT PROCESSING",
  PROCESSING = "PROCESSING",
  DONE = "DONE",
  ERROR = "ERROR",
}

export interface AttendanceState {
  status: AttendanceStatus;
  data: OriginAttendance | null;
}

const initialState: AttendanceState = {
  status: AttendanceStatus.NOT_PROCESSING,
  data: null,
};

export const getAttendanceExtraReducers = (
  builder: ActionReducerMapBuilder<AttendanceState>
) => {
  // 取得
  builder
    .addCase(fetchAttendance.pending, (state) => {
      state.status = AttendanceStatus.PROCESSING;
    })
    .addCase(fetchAttendance.fulfilled, (state, action) => {
      state.status = AttendanceStatus.DONE;
      state.data = action.payload;
    })
    .addCase(fetchAttendance.rejected, (state) => {
      state.status = AttendanceStatus.ERROR;
      state.data = null;
    });

  // 出勤
  builder
    .addCase(registerClockIn.pending, () => {})
    .addCase(registerClockIn.fulfilled, (state, action) => {
      state.data = action.payload;
    })
    .addCase(registerClockIn.rejected, (state) => {
      state.data = null;
    });

  // 退勤
  builder
    .addCase(registerClockOut.pending, () => {})
    .addCase(registerClockOut.fulfilled, (state, action) => {
      state.data = action.payload;
    })
    .addCase(registerClockOut.rejected, (state) => {
      state.data = null;
    });

  // 備考欄更新
  builder
    .addCase(updateRemarks.pending, () => {})
    .addCase(updateRemarks.fulfilled, (state, action) => {
      state.data = action.payload;
    })
    .addCase(updateRemarks.rejected, (state) => {
      state.data = null;
    });
};

const attendanceSlice = createSlice({
  name: "attendance",
  initialState,
  reducers: {},
  extraReducers: (builder) => getAttendanceExtraReducers(builder),
});

export default attendanceSlice.reducer;

export const testAttendanceSlice = (customInitialState: AttendanceState) =>
  createSlice({
    name: "attendance",
    initialState: customInitialState,
    reducers: {},
    extraReducers: (builder) => getAttendanceExtraReducers(builder),
  }).reducer;
