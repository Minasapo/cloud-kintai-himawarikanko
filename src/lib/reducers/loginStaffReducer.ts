import { ActionReducerMapBuilder, createSlice } from "@reduxjs/toolkit";
import { Staff } from "../../api";
import fetchStaff from "../staff/FetchStaff";

export enum StaffStatus {
  NOT_PROCESSING = "NOT_PROCESSING",
  PROCESSING = "PROCESSING",
  DONE = "DONE",
  ERROR = "ERROR",
}

export interface StaffState {
  status: StaffStatus;
  data?: Staff;
}

const initialState: StaffState = {
  status: StaffStatus.NOT_PROCESSING,
  data: undefined,
};

export const getStaffExtraReducers = (
  builder: ActionReducerMapBuilder<StaffState>
) => {
  builder
    .addCase(fetchStaff.pending, (state) => {
      state.status = StaffStatus.PROCESSING;
    })
    .addCase(fetchStaff.fulfilled, (state, action) => {
      state.status = StaffStatus.DONE;
      state.data = action.payload as Staff;
    })
    .addCase(fetchStaff.rejected, (state) => {
      state.status = StaffStatus.ERROR;
    });
};

const loginStaffReducer = createSlice({
  name: "loginStaff",
  initialState,
  reducers: {
    clearStaff(state) {
      state.status = StaffStatus.NOT_PROCESSING;
      state.data = undefined;
    },
  },
  extraReducers: (builder) => getStaffExtraReducers(builder),
});

export default loginStaffReducer.reducer;

export const { clearStaff } = loginStaffReducer.actions;

export const testLoginStaffReducer = (customInitialState: StaffState) =>
  createSlice({
    name: "loginStaff",
    initialState: customInitialState,
    reducers: {
      clearStaff(state) {
        state.status = StaffStatus.NOT_PROCESSING;
        state.data = undefined;
      },
    },
    extraReducers: (builder) => getStaffExtraReducers(builder),
  }).reducer;
