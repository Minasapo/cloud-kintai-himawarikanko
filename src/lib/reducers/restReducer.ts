import { ActionReducerMapBuilder, createSlice } from "@reduxjs/toolkit";

// eslint-disable-next-line import/no-cycle
import { RootState } from "../../app/store";
import fetchRest, { OriginRest } from "../time_record/FetchRest";
import registerRestEnd from "../time_record/RegisterRestEnd";
import registerRestStart from "../time_record/RegisterRestStart";

export enum RestStatus {
  NOT_PROCESSING = "NOT PROCESSING",
  PROCESSING = "PROCESSING",
  DONE = "DONE",
  ERROR = "ERROR",
}

export interface RestState {
  status: RestStatus;
  data: OriginRest | null;
}

const initialState: RestState = {
  status: RestStatus.NOT_PROCESSING,
  data: null,
};

export const getRestExtraReducers = (
  builder: ActionReducerMapBuilder<RestState>
) => {
  // 取得
  builder
    .addCase(fetchRest.pending, (state) => {
      state.status = RestStatus.PROCESSING;
    })
    .addCase(fetchRest.fulfilled, (state, action) => {
      state.status = RestStatus.DONE;
      state.data = action.payload;
    })
    .addCase(fetchRest.rejected, (state) => {
      state.status = RestStatus.ERROR;
      state.data = null;
    });

  // 休憩開始
  builder
    .addCase(registerRestStart.pending, (state) => {
      state.status = RestStatus.PROCESSING;
    })
    .addCase(registerRestStart.fulfilled, (state, action) => {
      state.status = RestStatus.DONE;
      state.data = action.payload;
    })
    .addCase(registerRestStart.rejected, (state) => {
      state.status = RestStatus.ERROR;
      state.data = null;
    });

  // 休憩終了
  builder
    .addCase(registerRestEnd.pending, (state) => {
      state.status = RestStatus.PROCESSING;
    })
    .addCase(registerRestEnd.fulfilled, (state, action) => {
      state.status = RestStatus.DONE;
      state.data = action.payload;
    })
    .addCase(registerRestEnd.rejected, (state) => {
      state.status = RestStatus.ERROR;
      state.data = null;
    });
};

const restSlice = createSlice({
  name: "rest",
  initialState,
  reducers: {},
  extraReducers: (builder) => getRestExtraReducers(builder),
});

export default restSlice.reducer;

export const testRestSlice = (customInitialState: RestState) =>
  createSlice({
    name: "rest",
    initialState: customInitialState,
    reducers: {},
    extraReducers: (builder) => getRestExtraReducers(builder),
  }).reducer;

export const selectRest = (state: RootState) => state.restReducer;
