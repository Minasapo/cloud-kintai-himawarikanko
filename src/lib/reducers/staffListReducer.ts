import { ActionReducerMapBuilder, createSlice } from "@reduxjs/toolkit";
import { Staff } from "../../api";
import fetchStaffList from "../staff/FetchStaffList";

export enum StaffListStatus {
  NOT_PROCESSING = "NOT_PROCESSING",
  PROCESSING = "PROCESSING",
  DONE = "DONE",
  ERROR = "ERROR",
}

export interface StaffListState {
  status: StaffListStatus;
  data: Staff[];
  selectedData?: Staff;
}

const initialState: StaffListState = {
  status: StaffListStatus.NOT_PROCESSING,
  data: [],
  selectedData: undefined,
};

export const getStaffListExtraReducers = (
  builder: ActionReducerMapBuilder<StaffListState>
) => {
  builder
    .addCase(fetchStaffList.pending, (state) => {
      state.status = StaffListStatus.PROCESSING;
    })
    .addCase(fetchStaffList.fulfilled, (state, action) => {
      state.status = StaffListStatus.DONE;
      state.data = action.payload as Staff[];
    })
    .addCase(fetchStaffList.rejected, (state) => {
      state.status = StaffListStatus.ERROR;
    });
};

const staffListReducer = createSlice({
  name: "staffList",
  initialState,
  reducers: {
    selectedStaff: (state, action) => {
      state.selectedData = action.payload as Staff;
    },
  },
  extraReducers: (builder) => getStaffListExtraReducers(builder),
});

export default staffListReducer.reducer;

export const { selectedStaff } = staffListReducer.actions;

export const testStaffListReducer = (customInitialState: StaffListState) =>
  createSlice({
    name: "staffList",
    initialState: customInitialState,
    reducers: {},
    extraReducers: (builder) => getStaffListExtraReducers(builder),
  }).reducer;
