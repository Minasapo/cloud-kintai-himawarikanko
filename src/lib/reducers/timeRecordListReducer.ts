import { ActionReducerMapBuilder, createSlice } from "@reduxjs/toolkit";
import fetchTimeRecordList, {
  TimeRecordList,
} from "../time_record_list/FetchTimeRecordList";

export enum TimeRecordListStatus {
  NOT_PROCESSING = "NOT PROCESSING",
  PROCESSING = "PROCESSING",
  DONE = "DONE",
  ERROR = "ERROR",
}

export interface TimeRecordListState {
  status: TimeRecordListStatus;
  data: TimeRecordList[];
}

const initialState: TimeRecordListState = {
  status: TimeRecordListStatus.NOT_PROCESSING,
  data: [],
};

const getTimeRecordListExtraReducers = (
  builder: ActionReducerMapBuilder<TimeRecordListState>
) => {
  builder
    .addCase(fetchTimeRecordList.pending, (state) => {
      state.status = TimeRecordListStatus.PROCESSING;
    })
    .addCase(fetchTimeRecordList.fulfilled, (state, action) => {
      state.status = TimeRecordListStatus.DONE;
      state.data = action.payload;
    })
    .addCase(fetchTimeRecordList.rejected, (state) => {
      state.status = TimeRecordListStatus.ERROR;
      state.data = [];
    });
};

const timeRecordListReducer = createSlice({
  name: "timeRecordList",
  initialState,
  reducers: {},
  extraReducers: (builder) => getTimeRecordListExtraReducers(builder),
});

export default timeRecordListReducer.reducer;

export const testTimeRecordListSlice = (
  customInitialState: TimeRecordListState
) =>
  createSlice({
    name: "timeRecordList",
    initialState: customInitialState,
    reducers: {},
    extraReducers: (builder) => getTimeRecordListExtraReducers(builder),
  }).reducer;
