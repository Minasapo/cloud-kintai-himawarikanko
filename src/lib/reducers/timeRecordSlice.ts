import { ActionReducerMapBuilder, createSlice } from "@reduxjs/toolkit";
// eslint-disable-next-line import/no-cycle
import getTimeRecordStatus from "../time_record/getTimeRecordStatus";

export enum TimeRecordStatus {
  PROCESSING = "PROCESSING",
  BEFORE_WORK = "BEFORE_WORK",
  WORKING = "WORKING",
  RESTING = "RESTING",
  LEFT_WORK = "LEFT_WORK",
  ERROR = "ERROR",
}

export const TimeRecordStatusText = {
  PROCESSING: "処理中",
  BEFORE_WORK: "出勤前",
  WORKING: "勤務中",
  RESTING: "休憩中",
  LEFT_WORK: "退勤済み",
  ERROR: "エラー",
};

export interface TimeRecordState {
  status: TimeRecordStatus;
  statusText: string;
}

const initialState: TimeRecordState = {
  status: TimeRecordStatus.BEFORE_WORK,
  statusText: TimeRecordStatusText.BEFORE_WORK,
};

export const getExtraReducers = (
  builder: ActionReducerMapBuilder<TimeRecordState>
) => {
  builder
    .addCase(getTimeRecordStatus.pending, (state) => {
      state.status = TimeRecordStatus.PROCESSING;
      state.statusText = TimeRecordStatusText.PROCESSING;
    })
    .addCase(getTimeRecordStatus.fulfilled, (state, action) => {
      state.status = action.payload;
      state.statusText = TimeRecordStatusText[action.payload];
    })
    .addCase(getTimeRecordStatus.rejected, (state) => {
      state.status = TimeRecordStatus.ERROR;
      state.statusText = TimeRecordStatusText.ERROR;
    });
};

const TimeRecordSlice = createSlice({
  name: "timeRecord",
  initialState,
  reducers: {},
  extraReducers: (builder) => getExtraReducers(builder),
});

export default TimeRecordSlice.reducer;

export const testTimeRecordSlice = (customInitialState: TimeRecordState) =>
  createSlice({
    name: "TimeRecord",
    initialState: customInitialState,
    reducers: {},
    extraReducers: (builder) => getExtraReducers(builder),
  }).reducer;
