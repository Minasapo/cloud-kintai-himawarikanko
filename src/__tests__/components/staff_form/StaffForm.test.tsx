import { ThemeProvider } from "@mui/material";
import { configureStore } from "@reduxjs/toolkit";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import StaffForm from "../../../components/staff_form/StaffForm";
import { AttendanceStatus, testAttendanceSlice } from "../../../lib/reducers/attendanceReducer";
import { LoginStaffStatus, testLoginStaffReducer } from "../../../lib/reducers/loginStaffReducer";
import { RestStatus, testRestSlice } from "../../../lib/reducers/restReducer";
import { StaffListStatus, testStaffListReducer } from "../../../lib/reducers/staffListReducer";
import { TimeRecordListStatus, testTimeRecordListSlice } from "../../../lib/reducers/timeRecordListReducer";
import { TimeRecordStatus, TimeRecordStatusText, testTimeRecordSlice } from "../../../lib/reducers/timeRecordSlice";
import { theme } from "../../../lib/theme";

const mockStore = configureStore({
  reducer: {
    timeRecordReducer: testTimeRecordSlice({
      status: TimeRecordStatus.PROCESSING,
      statusText: TimeRecordStatusText.PROCESSING,
    }),
    loginStaffReducer: testLoginStaffReducer({
      status: LoginStaffStatus.DONE,
      data: {
        staffId: 999,
        lastName: "田中",
        firstName: "太郎",
        mailAddress: "tanaka@example.com",
        iconPath: "",
        staffRoles: {
          roleId: 2,
          staffId: 999,
          role: {
            roleName: "スタッフ",
          },
        },
      },
    }),
    staffListReducer: testStaffListReducer({
      status: StaffListStatus.PROCESSING,
      data: [],
    }),
    attendanceReducer: testAttendanceSlice({
      status: AttendanceStatus.DONE,
      data: null,
    }),
    restReducer: testRestSlice({
      status: RestStatus.DONE,
      data: null,
    }),
    timeRecordListReducer: testTimeRecordListSlice({
      status: TimeRecordListStatus.DONE,
      data: [],
    }),
  },
});

describe("StaffForm Component(", () => {
  test.concurrent("レイアウト崩れが発生していないか(ビジュアルリグレッション)", () => {
    const { asFragment } = render(
      <Provider store={mockStore}>
        <ThemeProvider theme={theme}>
          <StaffForm />
        </ThemeProvider>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
