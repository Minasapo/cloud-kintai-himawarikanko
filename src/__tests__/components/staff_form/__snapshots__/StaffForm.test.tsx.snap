// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1okj3ks-MuiStack-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiStack-root css-zsbir3-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-tuk8tq-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              スタッフコード
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1az0k99-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id=":r0:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    スタッフコード
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiStack-root css-u10pvi-MuiStack-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-tuk8tq-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r1:"
                  id=":r1:-label"
                >
                  名前(姓)
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-5v9kzl-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1az0k99-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="last-name"
                    id=":r1:"
                    maxlength="10"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        名前(姓) *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-tuk8tq-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r2:"
                  id=":r2:-label"
                >
                  名前(名)
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-5v9kzl-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1az0k99-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="first-name"
                    id=":r2:"
                    maxlength="10"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        名前(名) *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-tuk8tq-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r3:"
              id=":r3:-label"
            >
              メールアドレス
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-5v9kzl-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1az0k99-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="mail-address"
                id=":r3:"
                maxlength="100"
                pattern="^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+.[a-zA-Z]{2,}$"
                required=""
                type="email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    メールアドレス *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-tq4joq-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              権限
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-5v9kzl-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-ynug2-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-testid="staff-role"
              required=""
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="staff-role"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="button"
                tabindex="0"
              >
                スタッフ
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                required=""
                tabindex="-1"
                value="2"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    権限 *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-htqi8w-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            作成
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
